sdk/src/store.rs

```diff
@@ -1386,26 +1401,76 @@ impl Store {
 
         Ok(hashes)
     }
+
+    // move or copy data from source to dest
+    #[cfg(feature = "file_io")]
+    fn move_or_copy(source: &Path, dest: &Path) -> Result<()> {
+        // copy temp file to asset
+        std::fs::rename(source, dest)
+            // if rename fails, try to copy in case we are on different volumes or output does not exist
+            .or_else(|_| std::fs::copy(source, dest).and(Ok(())))
+            .map_err(Error::IoError)
+    }
+
+    // copy output and possibly the external manifest to final destination
+    #[cfg(feature = "file_io")]
+    fn copy_c2pa_to_output(source: &Path, dest: &Path, remote_type: RemoteManifest) -> Result<()> {
+        match remote_type {
+            crate::claim::RemoteManifest::NoRemote => Store::move_or_copy(source, dest)?,
+            crate::claim::RemoteManifest::SideCar | crate::claim::RemoteManifest::Remote(_) => {
+                // make correct path names
+                let source_asset = source;
+                let source_cai = source_asset.with_extension(MANIFEST_STORE);
+                let dest_cai = dest.with_extension(MANIFEST_STORE);
+
+                Store::move_or_copy(&source_cai, &dest_cai)?; // copy manifest
+                Store::move_or_copy(source_asset, dest)?; // copy asset
+            }
+        }
+        Ok(())
+    }
+
     /// Embed the claims store as jumbf into an asset. Updates XMP with provenance record.
     #[cfg(feature = "file_io")]
     pub fn save_to_asset(
         &mut self,
         asset_path: &Path,
         signer: &dyn Signer,
-        output_path: &Path,
-    ) -> Result<()> {
-        let jumbf_bytes = self.start_save(asset_path, output_path, signer.reserve_size())?;
+        dest_path: &Path,
+    ) -> Result<Vec<u8>> {
+        // set up temp dir, contents auto deleted
+        let td = tempfile::TempDir::new()?;
+        let temp_path = td.into_path();
+        let temp_file = temp_path.join(
+            dest_path
+                .file_name()
+                .ok_or_else(|| Error::BadParam("invalid destination path".to_string()))?,
+        );
+
+        let jumbf_bytes = self.start_save(asset_path, &temp_file, signer.reserve_size())?;
 
         let pc = self.provenance_claim().ok_or(Error::ClaimEncoding)?;
         let sig = self.sign_claim(pc, signer, signer.reserve_size())?;
         let sig_placeholder = self.sign_claim_placeholder(pc, signer.reserve_size());
 
-        match self.finish_save(jumbf_bytes, output_path, sig, &sig_placeholder) {
-            Ok(v) => {
+        // get correct output path for remote manifest
```

Comment:
I ran out of time, so I'll stop reviewing here and publish as-is. I'll pick up from here later.